src/options.rs

```diff
@@ -175,13 +167,14 @@ impl Options {
 		salt: &Salt,
 		version: Option<u32>,
 	) -> Result<()> {
-		let version = version.unwrap_or(CURRENT_VERSION);
-		let mut file = std::fs::File::create(path)?;
-		writeln!(file, "version={}", version)?;
-		writeln!(file, "salt={}", hex::encode(salt))?;
+		let mut metadata = vec![
+			format!("version={}", version.unwrap_or(CURRENT_VERSION)),
+			format!("salt={}", hex::encode(salt)),
+		];
 		for i in 0..self.columns.len() {
-			writeln!(file, "col{}={}", i, self.columns[i].as_string())?;
+			metadata.push(format!("col{}={}", i, self.columns[i].as_string()));
 		}
+		std::fs::write(path, metadata.join("\n"))?;
```

Comment:
Hmm, this is still not guaranteed to be atomic, is it? Looking at the documentaion, this calls `std::io::File::write` in a loop, which may result in a file written only partially. TBH I don't think this is likely to cause any issues in the real world. All of the metadata is likely to fit into a single 4kb IO page anyway.